### DISCUSSION TITLE
etcdDatabaseHighFragmentationRatio should change default rule

### DISCUSSION THREAD

**Discussion by zgfh** [etcd-io/etcd#18620] [Ideas]
https://github.com/etcd-io/etcd/blob/main/contrib/mixin/alerts/alerts.libsonnet#L222
I meet alert: etcdDatabaseHighFragmentationRatio

becuse the default quota space is 2G of etcd
![image](https://github.com/user-attachments/assets/c3600123-e548-4574-8c91-355f931f9814)

I this is ok if there just 50M fragmentation
should we change the rule to 1024M?  like this(1073741824=1024M):
```
(last_over_time(etcd_mvcc_db_total_size_in_use_in_bytes[5m]) / last_over_time(etcd_mvcc_db_total_size_in_bytes[5m])) < 0.5 and etcd_mvcc_db_total_size_in_use_in_bytes > 1073741824
```

the default alert is 
```
  (last_over_time(etcd_mvcc_db_total_size_in_use_in_bytes[5m]) / last_over_time(etcd_mvcc_db_total_size_in_bytes[5m])) < 0.5 and etcd_mvcc_db_total_size_in_use_in_bytes > 104857600

```

link: https://github.com/prometheus-operator/kube-prometheus/issues/2470

**Comment by jmhbnz** [comment]
Hi @zgfh - Thanks for raising this idea.  We previously merged a change related to this in https://github.com/etcd-io/etcd/pull/15291.  Did we miss something in that pr or is the change not working as expected?

**Reply by zgfh** [reply]
Thanks for your help
 I think it would be better if the metrics are ignored unless total database size is 1024MB+

**Reply by LTP7534** [reply]
I also think that the etcd_mvcc_db_total_size_in_use_in_bytes setting to 100 MB is too sensitive.
It is easy to trigger alarms in some large clusters and we are not very willing to perform debris cleanup just because the debris space may have reached 100 MB

**Reply by ahrtr** [reply]
Note that there are two conditions, and there is an **AND**. The alert will only be triggered when both are true.
 - (last_over_time(etcd_mvcc_db_total_size_in_use_in_bytes[5m]) / last_over_time(etcd_mvcc_db_total_size_in_bytes[5m])) < 0.5  
- etcd_mvcc_db_total_size_in_use_in_bytes > 104857600

**Reply by jadolg** [reply]
@ahrtr Would it make sense to use `etcd_server_quota_backend_bytes / 2` if we're not using the default quota given that this number is technically half that?
